kedro_azureml/datasets/folder_dataset.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from typing import Any, Dict, Optional, Type, Union
+
+from kedro.io.core import (
+    VERSION_KEY,
+    VERSIONED_FLAG_KEY,
+    AbstractDataSet,
+    DataSetError,
+    Version,
+)
+
+from kedro_azureml.datasets.pipeline_dataset import AzureMLPipelineDataSet
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMLFolderDataSet(AzureMLPipelineDataSet):
+    def __init__(
+        self,
+        azureml_dataset: str,
+        dataset: Union[str, Type[AbstractDataSet], Dict[str, Any]],
+        version: Optional[Version] = None,
+        folder: str = "data",
```

Comment:
Maybe rename `folder` to `root_dir`? Could you explain why this param was introduced both here and in the base `AzureMLPipelineDataSet`?

Reply:
`root_dir` is a bit clearer indeed. It was introduced to support two cases that don't work with the current approach:
1. Files not in the root of a dataset
2. Kedro datasets that use folders instead of a single file

While 1. is probably not important to have when passing data using `AzureMLPipelineDataSet`, 2. would be useful IMO, so therefore it has been added to `AzureMLPipelineDataSet` as well.